public/content/en/gems/ndslice.md

```diff
@@ -0,0 +1,120 @@
+# Multidimensional arrays
+
+D provides support for multidimensional arrays directly in its standard library
+Phobos. It allows fast iteration, access and manipulation of multidimensional
+data. In comparison to popular software packages in other languages (e.g. `ndarray`
+in NumPy or `Array` in Julia), it doesn't become slow for computations not
+covered by preexisting functionality in a bound C-library nor forces the user
+to escape into another language. Additionally, due to its compile-time introspection
+capabilities, D enables further optimizations for superior performance.
+
+## Slices
+
+A `Slice` is a view on a multidimensional view on usually an underlying raw,
+one-dimensional array. With the `sliced` function a multidimensional array can
+be constructed - in this example a `2x3` matrix is created:
+
+```d
+auto arr = new double[6];       // allocate one-dimensional array
+auto matrix = arr.sliced(2, 3); // construct view onto array
+```
+
+As specifying the size of the to-be-allocated array is tedious, `ndslice` provides
+a `slice` function, which will automatically allocate:
+
+```d
+auto matrix = slice!int(2, 3);
+```
+
+### Row/column styles
+
+For the more common row-first index-style (C, D) square brackets can be used `a[]`,
+whereas the column-first index style (Matlab, Fortran) uses parenthesis `a()`.
+A row or column is also a `Slice` and thus all operations that work on slices
+work on selections, for example we can assign a value or values.
+
+```d
+// [0, 1, 2]
+// [3, 4, 5]
+// [6, 7, 8]
+auto matrix = iotaSlice(3, 3).slice; // `slice` allocates from the iotaSlice
+matrix[1]   = [0, 1, 0]; // row-access, [3, 4, 5]
+matrix(1)[] = 42; // column-acces, [1, 4, 7]
+```
+
+For element access the same style exists:
+
+```d
+matrix[1, 0] = 42; // column-first, has value `3` before
+matrix(1, 0) = 42; // row-first,    has value `1` before
+```
+
+### Selections
+
+There's a [plethora](http://dlang.org/phobos/std_experimental_ndslice_selection.html)
```

Comment:
This is not plethora! Comparing with many other libraries for multidimensional arrays ndslice provide only small set of functions. In the same time their combination covers much more functionality. `pack` and friends should be explained